python/dgl/nn/pytorch/graph_transformer.py

```diff
@@ -219,3 +223,113 @@ def forward(self, ndata, attn_bias=None, attn_mask=None):
         attn = self.out_proj(attn.reshape(N, bsz, self.feat_size).transpose(0, 1))
 
         return attn
+
+
+class GraphTransformerLayer(nn.Module):
+    r"""Graph Transformer Layer with Dense Multi-Head Attention, as introduced
+    in `Do Transformers Really Perform Bad for Graph Representation?
+    <https://arxiv.org/pdf/2106.05234>`__
+
+    Parameters
+    ----------
+    feat_size : int
+        Feature size.
+    ffn_size : int
+        Hidden size of feedforward layers.
+    num_heads : int
+        Number of attention heads, by which attr:`feat_size` is divisible.
+    attn_bias_type : str, optional
+        The type of attention bias used for modifying attention. Selected from
+        'add' or 'mul'. Default: 'add'.
+
+        * 'add' is for additive attention bias.
+        * 'mul' is for multiplicative attention bias.
+    norm_first : bool, optional
+        If True, it performs layer norm before attention and feedforward
+        operations. Otherwise, it applies layer norm after. Default: False.
+    dropout : float, optional
+        Dropout probability. Default: 0.1.
+    activation : str, optional
+        Activation function. Default: 'relu'.
+
+    Examples
+    --------
+    >>> import torch as th
+    >>> from dgl.nn import GraphTransformerLayer
+
+    >>> ndata = th.rand(16, 100, 512)
+    >>> bias = th.rand(16, 100, 100, 8)
+    >>> net = GraphTransformerLayer(feat_size=512, ffn_size=2048, num_heads=8)
+    >>> out = net(ndata, bias)
+    """
+
+    def __init__(
+        self,
+        feat_size,
+        ffn_size,
+        num_heads,
+        attn_bias_type='add',
+        norm_first=False,
+        dropout=0.1,
+        activation='relu'
+    ):
+        super().__init__()
+
+        self.norm_first = norm_first
+
+        self.attn = BiasedMultiheadAttention(
+            feat_size=feat_size,
+            num_heads=num_heads,
+            attn_bias_type=attn_bias_type,
+            attn_drop=dropout
+        )
+        self.fc1 = nn.Linear(feat_size, ffn_size)
+        self.fc2 = nn.Linear(ffn_size, feat_size)
+
+        self.dropout = nn.Dropout(p=dropout)
+        self.activation_dropout = nn.Dropout(p=dropout)
+        self.activation = getattr(F, activation)
+        self.attn_layer_norm = nn.LayerNorm(feat_size)
+        self.ffn_layer_norm = nn.LayerNorm(feat_size)
+
+    def forward(self, ndata, attn_bias=None, attn_mask=None):

```

Comment:
Maybe use `feat` or `nfeat`, which has been widely used in the existing DGL modules